### ISSUE TITLE
Ordering operators for Addr# don't work

### ISSUE THREAD

**Issue by ryantrinkle** [ghcjs/ghcjs#608]
[`AddrGtOp`, `AddrGeOp`](https://github.com/ghcjs/ghcjs/blob/eb8670c4b03b4a0ff5dd7ac3f524b410b37051ea/src/Gen2/Prim.hs#L402-L403), `AddrLtOp`, and `AddrLeOp` don't work, because javascript objects aren't in fact ordered.  I can't think of any reasonably-performant way to fix this, but I think it might make sense to do something about this - perhaps throw an exception, or refuse to generate these primops?

**Comment by hamishmack** [comment]
I think this should be fixed now.  The new h$comparePointer function is [here](https://github.com/ghcjs/shims/commit/78e2967c791ef29f0f752761a0cd0403ba89777e).

It gives an id to each new buffer it sees and it adds the `.dv.byteOffset` to identify the correct location in the buffer itself.

**Comment by ryantrinkle** [comment]
@hamishmack Wouldn't ghcjs need to be modified to use this in its primitives?

**Comment by hamishmack** [comment]
> @hamishmack Wouldn't ghcjs need to be modified to use this in its primitives?

Yes.  That part is [here](https://github.com/ghcjs/ghcjs/commit/c675302dc1bd52cc21a7a92f2854d5eda5ea2163).
